activerecord/lib/active_record/attribute_methods/query.rb

```diff
@@ -16,9 +16,8 @@ def query_attribute(attr_name)
         when true        then true
         when false, nil  then false
         else
-          column = self.class.columns_hash[attr_name]
-          if column.nil?
-            if Numeric === value || value !~ /[^0-9]/
+          if !self.type_for_attribute(attr_name) { false }
```

Comment:
Is it possible that `!self.type_for_attribute(attr_name) { false }` returns true? I do not understand when that will be happen.

Reply:
If a result of any queries has missing attribute name in `attribute_types` on the model, this condition returns true.